remix/app/routes/_app.newsletter.update.$id.tsx

```diff
@@ -0,0 +1,88 @@
+import { useEffect } from 'react';
+
+import { ActionArgs, json } from '@remix-run/node';
+import { Form, useActionData, useNavigate } from '@remix-run/react';
+import { withZod } from '@remix-validated-form/with-zod';
+import { validationError } from 'remix-validated-form';
+import { z } from 'zod';
+
+import { addNotification } from '../helpers/localStorage.helper';
+import appHandler from '../lib/handler/app.handler';
+import NewsletterRepository from '../repositories/newsletter.server';
+
+const validator = withZod(
+  z.object({
+    name: z.string().min(1, { message: 'Name is required' }),
+    content: z.string().min(1, { message: 'Content is required' }),
+  })
+);
+
+export async function action({ request, params }: ActionArgs) {
+  return appHandler(request, async (user) => {
+    const result = await validator.validate(await request.formData());
+
+    if (result.error) {
+      return validationError(result.error);
+    }
+
+    const newsletterExists = await NewsletterRepository.findOne({
+      where: { id: params.id },
+    });
+
+    if (!newsletterExists) {
+      return validationError({
```

Comment:
We should instead return a 404 page: https://remix.run/docs/en/main/guides/not-found#how-to-send-a-404. Landing on a page with an error message is not great 😅

Reply:
Yep! I have improved here: [d426473](https://github.com/nimblehq/js-framework-benchmark/pull/126/commits/d426473d63ea5e96d4c67a790b23169b39f78847)